sklearn/linear_model/tests/test_logistic.py

```diff
@@ -799,25 +799,34 @@ def test_ovr_multinomial_iris():
 
 
 def test_logistic_regression_solvers():
+    """
+    Test that available logistic solvers converge to the same result on a fixed
+    synthetic dataset. The solvers are compared unordered pairwise omitting self-
+    comparisons.
+    """
     X, y = make_classification(n_features=10, n_informative=5, random_state=0)
 
     params = dict(fit_intercept=False, random_state=42, multi_class="ovr")
     solvers = ("newton-cg", "lbfgs", "liblinear", "sag", "saga", "trust-ncg")
 
-    # dict of fitted logistic regressors
     regressors = {
         solver: LogisticRegression(solver=solver, **params).fit(X, y)
         for solver in solvers
     }
 
-    # test over unordered pairs of distinct solvers
     for solver_1, solver_2 in itertools.combinations(regressors):
         assert_array_almost_equal(
             regressors[solver_1].coef_, regressors[solver_2].coef_, decimal=3
         )
 
 
 def test_logistic_regression_solvers_multiclass():
+    """
+    Test that available logistic solvers converge to the same result on a fixed
+    synthetic dataset. The solvers are compared unordered pairwise omitting self-
+    comparisons. Specific solvers have increased `max_iter` values to allow for
+    proper convergence.
+    """
```

Comment:
Same comment here regarding pairwise convergence. The specific solver part can be moved down near the implementation.

```suggestion
    """Test solvers converge to the same result for multiclass problems."""
```

I think of the docstring for test as "short and high level overview" and inline comments are used if something is not obvious by looking at the code. In the end, inline comments are a judgement call. For me  adjusting `max_iter` is "a little less obvious".